pilot/pkg/dns/monitoring.go

```diff
@@ -34,16 +36,39 @@ var (
 		"Total number of DNS requests forwarded to upstream.",
 	)
 
-	requestDuration = monitoring.NewDistribution(
+	upstreamRequestDuration = monitoring.NewDistribution(
 		"dns_upstream_request_time_ms",
 		"Total time in milliseconds Istio takes to get DNS response from upstream.",
 		[]float64{1, 3, 5, 10, 25, 50},
 	)
+
+	requestDuration = monitoring.NewDistribution(
+		"dns_request_time_ms",
+		"Total time in milliseconds Istio takes to serve DNS response.",
+		[]float64{1, 3, 5, 10, 25, 50},
+	)
+
+	rcodeLabel = monitoring.MustCreateLabel("rcode")
+
+	responses = monitoring.NewSum(
+		"dns_responses_total",
+		"Total number of DNS responses.",
+		monitoring.WithLabels(rcodeLabel),
+	)
 )
 
 func registerStats() {
 	monitoring.MustRegister(requests)
 	monitoring.MustRegister(upstreamRequests)
 	monitoring.MustRegister(failures)
-	monitoring.MustRegister(requestDuration)
```

Comment:
requestDuration metric is not registered.

Reply:
totally forgot about these - done